code/graphics/opengl/gropenglshader.cpp

```diff
@@ -336,6 +336,10 @@ static SCP_string opengl_shader_get_header(shader_type type_id, int flags, shade
 		sflags << "#define WORKAROUND_CLIPPING_PLANES\n";
 	}
 
+	if (Detail.lighting < 3) {
```

Comment:
Do shaders get recompiled if the detail level is changed? If not then this would lead to inconsistent lighting effects until the game is restarted. Maybe a `gr_` function should be added which recompiles all existing shaders.

Reply:
That's a good point, a shader recompile definitely needs to happen here.